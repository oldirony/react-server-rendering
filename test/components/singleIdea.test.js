import React from 'react';
import SingleIdea from '../../src/components/single-idea/single-idea';

const props = {
	title: 'Title',
	timestamp: "2017-05-31T11:56:27.286Z",
	sections: []
}

test('It renders correctly', ()=>{
	let form = shallow(<SingleIdea {...props}/>);
	expect(form).toMatchSnapshot();
})
test('It renders correctly with content', ()=>{
	const content = `
		row 1
		row 2
	`;

	let form = shallow(<SingleIdea {...props} content={content}/>);
	expect(form).toMatchSnapshot();
})