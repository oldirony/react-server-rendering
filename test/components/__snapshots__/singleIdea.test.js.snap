// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<div
  className="idea"
>
  <h2
    className="title"
  >
    Title
  </h2>
  <time
    className="date"
  >
    12:56 
    31-4-2017
  </time>
  <ul
    className="sections"
  />
  <nav
    className="nav"
  >
    <Link
      className="button"
      replace={false}
      to="/ideas/:id/edit"
    >
      Edit
    </Link>
    <Link
      className="button"
      replace={false}
      to="/ideas"
    >
      Close
    </Link>
  </nav>
</div>
`;

exports[`It renders correctly with content 1`] = `
<div
  className="idea"
>
  <h2
    className="title"
  >
    Title
  </h2>
  <time
    className="date"
  >
    12:56 
    31-4-2017
  </time>
  <div
    className="content"
  >
    <p>
      row 1
    </p>
    <p>
      		row 2
    </p>
  </div>
  <ul
    className="sections"
  />
  <nav
    className="nav"
  >
    <Link
      className="button"
      replace={false}
      to="/ideas/:id/edit"
    >
      Edit
    </Link>
    <Link
      className="button"
      replace={false}
      to="/ideas"
    >
      Close
    </Link>
  </nav>
</div>
`;
